src/index.ts

```diff
@@ -364,19 +380,46 @@ export default function fetchHAR(har: Har, opts: FetchHAROptions = {}) {
         options.body = request.postData.text;
       }
     }
+
+    /**
+     * The fetch spec, which Node 18+ strictly abides by, now requires that `duplex` be sent with
+     * requests that have payloads.
+     *
+     * As `RequestInit#duplex` isn't supported by any browsers, or even mentioned on MDN, we aren't
+     * sending it in browser environments. This work is purely to support Node 18+ and `undici`
+     * environments.
+     *
+     * @see {@link https://github.com/nodejs/node/issues/46221}
+     * @see {@link https://github.com/whatwg/fetch/pull/1457}
+     * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/Request/Request}
+     */
+    if (shouldSetDuplex && !isBrowser()) {
+      options.duplex = 'half';
+    }
   }
 
   // We automaticaly assume that the HAR that we have already has query parameters encoded within
   // it so we do **not** use the `URLSearchParams` API here for composing the query string.
+  let requestURL = url;
   if ('queryString' in request && request.queryString.length) {
-    const urlObj = new URL(url);
+    const urlObj = new URL(requestURL);
 
     const queryParams = Array.from(urlObj.searchParams).map(([k, v]) => `${k}=${v}`);
     request.queryString.forEach(q => {
       queryParams.push(`${q.name}=${q.value}`);
     });
 
     querystring = queryParams.join('&');
+
+    // Because anchor hashes before query strings will prevent query strings from being delivered
+    // we need to pop them off and re-add them after.
+    const urlHashes = url.split('#');
+    requestURL = `${urlHashes[0].split('?')[0]}${querystring ? `?${querystring}` : ''}`;
+    if (urlHashes.length > 1) {
+      // Though web servers don't have access to hashes we might as well send it anyways as it
+      // was intentionally added to the URL.
+      requestURL += `#${urlHashes.join('#')}`;
+    }
```

Comment:
Amusingly [httpsnippet](https://npm.im/httpsnippet) is able to render code snippets with URLs that have hashes + query params but it's doing so by using the `url.format` library in Node. As `fetch-har` is intended to be cross-platform and I've tried to limit the amount of external libraries required to run it I'm opting to manually handle this edge case here instead of importing the `url` library.